src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -792,12 +792,17 @@ bool async_manifest_view::in_stm(async_view_search_query_t o) {
           return ko >= sko;
       },
       [this](model::timestamp ts) {
-          auto sm = _stm_manifest.timequery(ts);
-          if (!sm.has_value()) {
-              return false;
+          vlog(_ctxlog.debug, "Checking timestamp {} using timequery", ts);
+          if (_stm_manifest.get_spillover_map().empty()) {
+              // The STM manifest is empty, so the timestamp has to be directed
+              // to the STM manifest.
+              // Implicitly, this case also handles the empty manifest case
+              // because the STM manifest with spillover segments is never
+              // empty.
+              return true;
```

Comment:
This function is called `in_stm`, presumably testing if the query should be handled by the stm. However, we return `true` here when `The STM manifest is empty`. Perhaps it's only a wording issue, but if the stm is empty, how would anything be in it?

Reply:
I think the comment meant to say: `We have no spillover manifests, so...`.

Reply:
If it's empty then all queries has to be handled by the stm (which will handle this particular corner case well). The archive can't exist if the manifest is empty